R/client-cloud.R

```diff
@@ -66,20 +66,46 @@ cloudClient <- function(service, authInfo) {
       listRequest(service, authInfo, path, query, "applications")
     },
 
-    getApplication = function(applicationId) {
-      path <- paste("/applications/", applicationId, sep = "")
-      application <- GET(service, authInfo, path)
+    getApplication = function(outputOrApplicationId, deploymentRecordVersion) {
+      if (is.na(deploymentRecordVersion)) {
+        # In pre-versioned dcf files, outputOrApplicationId is the id of the application.
+        # TODO: consider removing support for this case a year after the release of 0.8.30
```

Comment:
It's never quite "safe" to remove this block – as long as someone has an old has an old deployment DCF file, this would break for them on removal. We don't have visibility into that, but I'm assuming we don't want to support this case forever.